C++/test/algorithm/dynamic_programming/longest_increasing_subsequence.cpp

```diff
@@ -0,0 +1,13 @@
+#include "third_party/catch.hpp"
+#include "algorithm/dynamic_programming/longest_increasing_subsequence.hpp"
+
+TEST_CASE("Base Cases", "[LIS]") {
+  REQUIRE(lis(std :: vector < int > ({10, 22, 9, 33, 21, 50, 41, 60}),8) == 5);
```

Comment:
This isn't a base case. Please move this under "General cases" and add the following cases here:
* `{}` (empty vector)
* `{0}`
* `{1}`
* `{-1}`
* `{0, 1}`
* `{1, 0}`
* `{-1, 0}`
* `{0, -1}`
* `{-1, 0, 1}` and a few other permutations of it.